spec/integration/client_spec.rb

```diff
@@ -60,6 +60,30 @@
     end
   end
 
+  describe "#on_row_event, #on_save_position" do
+    context "when a row is inserted" do
+      let(:events_to_wait_for) { 4 }
+      let(:actual_event_order) do
+        event_order = []
+        subject.on_row_event { |_| event_order << :on_row_event }
+        subject.on_save_position { |_, _| event_order << :on_save_position }
+
+        subject.start_in_thread
+
+        DatabaseHelper.insert(table_name, mysql_row)
+
+        sleep 0.1 while event_order.count < events_to_wait_for
+        subject.stop
+        event_order.last(2)
```

Comment:
Perhaps a comment why we only care about the last 2 (and why we need to wait 4 times).